### DISCUSSION TITLE
Strongswan is dropping IKE_SA_INIT response packets

### DISCUSSION THREAD

**Discussion by sridharkondeti** [strongswan/strongswan#1816] [Q&A] [answered]
**System (please complete the following information):**
 - OS: Ubuntu 22.04
 - Kernel version (if applicable): 5.15
 - strongSwan version(s): 5.9.5
 - Tested/confirmed with the latest version: yes

**Describe the bug**
Strongswan is dropping response IKE_SA_INIT packets.

After rebooting the instance, we are unable to establish the IPSec connection. Charon logs shows that IKE_SA_INIT packets are being sent to the remote server, and tcpdump shows that the response IKE_SA_INIT packets received properly, but somehow strongswan is not picking this response packets and retransmitting the packets and failing finally.

14[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
14[NET] sending packet: from <local>[500] to <remote>[500] (304 bytes)
07[IKE] retransmit 1 of request with message ID 0
07[NET] sending packet: from <local>[500] to <remote>[500] (304 bytes)
15[IKE] retransmit 2 of request with message ID 0
15[NET] sending packet: from <local>[500] to <remote>[500] (304 bytes)
13[IKE] retransmit 3 of request with message ID 0
13[NET] sending packet: from <local>[500] to <remote>[500] (304 bytes)
11[IKE] retransmit 4 of request with message ID 0
11[NET] sending packet: from <local>[500] to <remote>[500] (304 bytes)
15[IKE] retransmit 5 of request with message ID 0
15[NET] sending packet: from <local>[500] to <remote>[500] (304 bytes)
07[IKE] giving up after 5 retransmits
07[IKE] establishing IKE_SA failed, peer not responding

If we restart the strongswan service it works fine.

Can we fix this issue without restarting the strongswan service?

**To Reproduce**
1. Establish the connection
2. Reboot the VM
3. Try establishing the connection again.

This issue is intermittent.

**Comment by tobiasbrunner** [comment]
You might be running multiple IKE daemons. Make sure only charon or charon-systemd is running, not both.

**Reply by sridharkondeti** [reply]
Thanks @tobiasbrunner for the response. It always runs both charon and charon-systemd by default. I have few other VMs, there both daemons are running and no problem happened.
I checked daemons/configs/ports etc, everything looks same on both working instance and failed instance. Even the tcpdump messages looks same both Out and In messages.

**Reply by tobiasbrunner** [reply, marked as answer]
If you run both daemons, it's pure luck whether it works as both will open and listen on the same UDP ports (since [5.9.8](https://github.com/strongswan/strongswan/releases/tag/5.9.8) only one daemon can open them, so there you'd get an error when a second daemon is started). Make sure you only run one daemon by disabling the systemd unit for the one you don't want to use (`strongswan-starter` is the unit for starter/charon, `strongswan` the one for charon-systemd). You can also uninstall the package of the daemon you don't use (starter/charon is provided by the `strongswan-starter` and `strongswan-charon` packages, while charon-systemd is provided by `charon-systemd`).
